File: Lab03/ejercicio_propuesto/python/chat_message.py
class ChatMessage:

    WHOISIN = 0
    MESSAGE = 1
    LOGOUT = 2

    PRIVATE = 3
    FUNCTION = 4
    FILE = 5

    def __init__(self, msg_type, message, extra=None):
        self.type = msg_type
        self.message = message
        self.extra = extra

    def get_type(self):
        return self.type

    def get_message(self):
        return self.message

    def get_extra(self):
        return self.extra

File: Lab03/ejercicio_propuesto/java/ChatMessage.java
package Lab03.ejercicio_propuesto.java;


import java.io.*;

/*
 * This class defines the different type of messages that will be exchanged between the
 * Clients and the Server.
 */
public class ChatMessage implements Serializable {

    // Tipos de mensajes
    static final int WHOISIN = 0;
    static final int MESSAGE = 1;
    static final int LOGOUT = 2;

    private int type;
    private String message;

    // Constructor
    ChatMessage(int type, String message) {
        this.type = type;
        this.message = message;
    }

    int getType() {
        return type;
    }

    String getMessage() {
        return message;
    }
}